backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraTestRunner.java

```diff
@@ -17,14 +17,33 @@
  * under the License.                                           *
  ****************************************************************/
 
-package org.apache.james.mailbox.cassandra.mail;
+package org.apache.james.backends.cassandra;
 
-import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
-public class CassandraMailboxPathV2DAOTest extends CassandraMailboxPathDAOTest {
+public class CassandraTestRunner {
+    protected static DockerCassandraRule dockerCassandra = new DockerCassandraRule();
 
-    @Override
-    CassandraMailboxPathDAO testee() {
-        return new CassandraMailboxPathV2DAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+    static { // BeforeAll is played for each inner class
+        dockerCassandra.start();
+        // Rely on test containers for automated test cleanup
     }
-}
\ No newline at end of file
+
+    public abstract class Runner {
```

Comment:
Why it's called a Runner ? It looks like it runs nothing, AFAIU it does setup/teardown and provide a CassandraModule to setup the db. What about CassandraDatabaseInitializer ? And do we really need inheritance for that ?

Reply:
A registered code extension would be better, agree, but sadly not available on our JUNIT version. Note that next JUNIT versions skips tests, and we already rolled back from this, so I did not wanted to give it a go.

Reply:
I don't really understand what you are talking about and how it's related to my questions

Reply:
I was answering the "And do we really need inheritance for that ?"

I can very well do that in each test, however. Just that I'm lazy, I don't like repetitive code...